provider/awssd/aws_sd.go

```diff
@@ -409,23 +409,20 @@ func (p *AWSSDProvider) ListServicesByNamespaceID(namespaceID *string) (map[stri
 	return servicesMap, nil
 }
 
-// ListInstancesByServiceID returns list of instances registered in given service.
-func (p *AWSSDProvider) ListInstancesByServiceID(serviceID *string) ([]*sd.InstanceSummary, error) {
-	instances := make([]*sd.InstanceSummary, 0)
+// DiscoverInstancesByServiceID returns list of instances registered in given service.
+func (p *AWSSDProvider) DiscoverInstancesByServiceName(namespaceName, serviceName *string) ([]*sd.HttpInstanceSummary, error) {
```

Comment:
Why is this public?

Reply:
good point, after review I found that there are actually no custom logic left there, just a 1to1 wrapper around aws sdk client method so I removed it in favor of direct method call.